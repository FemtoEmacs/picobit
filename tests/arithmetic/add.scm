(define-macro (test-cond condition name) ;; TODO watch out for name capture
  `(let ((DUMMY-x ,condition))
     (display ,name)
     (if DUMMY-x
       (display " : PASSED\n")
       (begin (display " : FAILED, got : ")
	      (display DUMMY-x)
	      (display "\n")))))

(test-cond (= (+ 2 2) 4) "2 + 2 = 4")
(test-cond (= (+ 0 5) 5) "0 + 5 = 5")
(test-cond (= (+ 2 7) 9) "2 + 7 = 9")
(test-cond (= (+ 0 -5) -5) "0 + -5 = -5")
(test-cond (= (+ 2 -5) -3) "2 + -5 = -3")
(test-cond (= (+ 7 -5) 2) "7 + -5 = 2")
(test-cond (= (+ -3 -7) -10) "-3 + -7 = -10")

(test-cond (= (+ 100000 1) (+ 50000 50001)) "100000 + 1 = 50000 + 50001")
(test-cond (= (+ 32768 32768) 65536) "32k + 32k = 64k")
