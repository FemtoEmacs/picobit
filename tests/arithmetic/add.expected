:0080000080
:10800000FBD7BC0281AE4000815B4000816A40002A
:108010000003FFFB0003FFFD0004800081704000AF
:10802000000586A0817D4000000500008191400090
:1080300081954000819E40000004C3510003FFF978
:1080400081B2400081B340000004C3508159400018
:10805000816440008168400081364000813E4000DC
:108060000003FFF6812C4000812E40008066016FE6
:1080700080240171802401728024017380240174A2
:10808000806F01758031017680340177807D0178C1
:10809000802F01798024017A8041017B8034017C2A
:1080A000803B017E8035017F8041018080340181E9
:1080B0008031018280780002802401838078018CE5
:1080C00080240184802401858057018680360187C1
:1080D00080240188803701898024018A8037018BC0
:1080E0008070018C8041018D8050018E8036018F1F
:1080F00080240190803D000280340192806F000254
:108100008031019380240194802401968069019735
:1081100080240198802401998045019A80390002C9
:108120008073019B802F019C8024019D8066019FAC
:10813000803901A0804501A180360002802401A27F
:10814000803201A3804101A4803401A5803A01A6B8
:10815000802F01A7802401A8802701A9802401AADB
:10816000802401AB803901AC803401AD804801AE80
:10817000802F01AF806F01B080370002802F01B1E6
:10818000804801B4803701B5802F01B6803401B733
:10819000804001B8803401B9802401BA802401BB39
:1081A000802401BC803501BD804001BE802401BF18
:1081B000806E0146803B014780350148805401496B
:1081C0008031014A806B014B8024014C8039014D84
:1081D000803500028071014E8024014F804A015099
:1081E00080240151803401528024015380240154A1
:1081F000803401558024015680240157803B01586A
:10820000802F015A8031015C8039015D8049015E17
:108210008024015F803E01A38036016080310161CE
:1082200080240162804200028024014A8049016367
:1082300080240165803901668027016780340169E7
:108240008039016B8031016C8027016D8041016EA6
:1082500080670132802F013380410134805701351E
:108260008078013780240138804101398031013A1A
:108270008073013B8024013C804D013D802F013FF4
:108280008024000280240140803401418038014272
:108290008024014380410144806A00028036014508
:1082A000803E0128802401298034012A800E0002AA
:1082B0008024012B8024012D8024012F8027013070
:1082C0008036013180300122803601238024012450
:1082D000803401258077012680240127802F011F0B
:1082E000803E012080410121803400028073011E04
:1082F0008038000202B405505002B4054A51060607
:10830000C108C702B4052C61EE0409C109C702B453
:10831000050E61EE060BC10DC702B404F061EE0458
:10832000A107C1A107C702B404D261EE06A107C12B
:10833000A108C702B404B461EE0BA107C106C702CD
:10834000B4049661EEA108A112C1A11BC702B40436
:108350007861EEA10B05C1A115A111C1C702B4043A
:108360005A61EEA109A109C1A10DC702B40372614E
:10837000EED801A117B589EE20B3044AA106B68054
:108380000120DAB8954121DBB6800221CC9A21CEBA
:108390002161EE2022CFB67200EF20B6800120DAF4
:1083A0009EA02607E620B558EEA02607E600EF209F
:1083B000C0B8BF20B582B648012004C8B8AEA0310D
:1083C00021B5A402B583CDDCEF02A034210EC5C1D6
:1083D00022CD2102CDB403D97101210EC89220EF24
:1083E000210E406221B662010421C2EF2002B5597C
:1083F000B75520CCB8B8A02C07E620CEB51FEE208C
:10840000CFB6800120D296A02D07E600EF20CC9DAC
:10841000A02407E620CEB505EE20CFB666A10CB0AD
:108420000380EE20B0039DEEA02D07E600EF20D9DB
:1084300095A10EB1038020F09C2095A113B103807B
:10844000A116B10380A10FB10380A105B00380EE96
:1084500020B00380EEA104B1038001A11CB0038011
:10846000EE2095A106B10380A105B00380EE20B0F7
:108470000380EEA104B1038001A110B00380EE20BF
:1084800095A106B10380A105B00380EE20B0038062
:10849000EEA104B1038001A10AB00380EE2095A1F2
:1084A00006B10380A105B00380EE20B00380EEA1E9
:1084B00004B1038001A11AB00380EE2095A106B19A
:1084C0000380A105B00380EE20B00380EEA104B1CB
:1084D000038001A11DB00380EE2095A106B10380A9
:1084E000A105B00380EE20B00380EEA104B10380AB
:1084F00001A119B00380EE2095A106B10380A1056A
:10850000B00380EE20B00380EEA104B1038001A18E
:1085100018B00380EE2095A106B10380A105B00339
:1085200080EE20B00380EEA104B1038001A114B05D
:108530000380EE2095A106B10380A105B00380EE73
:1085400020B00380EEA104B10380012007E600EF14
:10855000022004C8212302CDCDB4055D710122049F
:10856000C822242302CDCDCDB4056C710123B8DC23
:1085700021B003E723222602CDCDCDB4057F7101C2
:1085800022B8C621B003E721242602CDCDCDB40503
:108590009271012022242602CDCDCDCDB405A0704C
:1085A000002121C4232502CDCDB405AD710122B82F
:1085B00093219220EF2298019420B103E720EF212C
:0F85C000B87BB77500B86EB76A21B73B21B725F6
:00000001FF
